### PR TITLE
Dockerfile: Remove not needed ENVs and add comment

### PR DESCRIPTION
**Pull request by casperklein**
# Description

This PR removes not needed ENVs and adds a comment to clarify, why the other ENVs are needed.

These ENVs are not needed, they are handled by `start-mailserver.sh` like all other ENVs in DMS:

- `ONE_DIR=1` 
- `ENABLE_POSTGREY`
- `SASLAUTHD_MECHANISMS`

The remaining ENVs are referenced in `target/supervisor/conf.d/saslauth.conf` and must be present when supervisord starts (tested).

If necessary, their values are adjusted by `start-mailserver.sh` on startup.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by georglauterbach — APPROVED** [review]
👍🏼

**Review by wernerfred — APPROVED** [review]
LGTM thx

File: Dockerfile [hunk 1]
```diff
@@ -21,14 +21,14 @@ LABEL org.opencontainers.image.url="https://github.com/docker-mailserver"
...
+# These ENVs are referenced in target/supervisor/conf.d/saslauth.conf
+# and must be present when supervisord starts.
+# If necessary, their values are adjusted by target/scripts/start-mailserver.sh on startup.
 ENV FETCHMAIL_POLL=300
-ENV ONE_DIR=1
 ENV POSTGREY_AUTO_WHITELIST_CLIENTS=5
 ENV POSTGREY_DELAY=300
 ENV POSTGREY_MAX_AGE=35
 ENV POSTGREY_TEXT="Delayed by Postgrey"
-ENV SASLAUTHD_MECHANISMS=pam
```
Comment (polarathene) [inline Dockerfile:31]: Just making a note that this ENV is not covered in `start-mailserver.sh` but in [`startup/setup-stack.sh`](https://github.com/docker-mailserver/docker-mailserver/blob/3cb5668b6f62258d8c21629c0f5e649085e4f0b6/target/scripts/startup/setup-stack.sh#L535), so it's not [persisted to our `/etc/dms-settings` file](https://github.com/docker-mailserver/docker-mailserver/blob/3cb5668b6f62258d8c21629c0f5e649085e4f0b6/target/scripts/startup/setup-stack.sh#L41) though I don't think that'd be an issue, if it were providing an explicit ENV to the container should resolve it?

**Review by polarathene — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (casperklein) [reply Dockerfile:31]: I noticed the same. There are other `SASLAUTHD_` ENVs that are also not covered by `start-mailserver.sh`:

https://github.com/docker-mailserver/docker-mailserver/blob/3cb5668b6f62258d8c21629c0f5e649085e4f0b6/target/scripts/startup/setup-stack.sh#L535-L541

`/etc/dms-settings` is only populated from ENVs defined in `start-mailserver.sh`. So removing this ENV from the Dockerfile is no regression.

**Pull request merged**